traj_analysis/traj_standalone.py

```diff
@@ -8,6 +8,108 @@
+    for i in range(dims):
+        # np.where neatly vectorizes the if-else logic for a 2D matrix
+        diff[:,i] = np.where(2*diff[:,i] > cell[i,i], cell[i,i] - diff[:,i], diff[:,i])
+    return diff
+
+def calc_onsagers(disp_list_1, disp_list_2, delta_t, nsamples):
+    """ Calculate onsager coefficients for a binary system to yield
+        a scalar for the Maxwell-Stefan diffusivity
+        Parameters
+        -----------
+        disp_list_1: list of ndarrays w/ shape (natoms,dims) if directional else (natoms,)
+        disp_list_2: list of ndarrays w/ shape (natoms,dims) if directional else (natoms,)
+        delta_t: float, time in picoseconds per traj frame
+        nsamples: int, number of samples for expectation values
+        -----------
+    """
+    ntotatoms = len(disp_list_1[0]) + len(disp_list_2[0])
+    onsager = np.zeros((2,2))
+    onsager_std = np.zeros((2,2))
+    # off-diagonal
+    onsager[0,1] = np.average(np.multiply([np.sum(disp_list_1[i]) for i in range(nsamples)],
+                                          [np.sum(disp_list_2[i]) for i in range(nsamples)]))/(delta_t*nsamples)/ntotatoms/6
+    # symmetry of onsager coefficients
+    onsager[1,0] = onsager[0,1]
+    # diagonal entries
+    onsager[0,0] = np.average(np.square([np.sum(disp_list_1[i]) for i in range(nsamples)]))/(delta_t*nsamples)/ntotatoms/6
+    onsager[1,1] = np.average(np.square([np.sum(disp_list_2[i]) for i in range(nsamples)]))/(delta_t*nsamples)/ntotatoms/6
+    # similar process for the error bounds on this measurement
+    onsager_std[0,1] = np.std(np.multiply([np.sum(disp_list_1[i]) for i in range(nsamples)],
+                                          [np.sum(disp_list_2[i]) for i in range(nsamples)]))/(delta_t*nsamples)/ntotatoms/6
+    onsager_std[1,0] = onsager_std[0,1]
+    onsager_std[0,0] = np.std(np.square([np.sum(disp_list_1[i]) for i in range(nsamples)]))/(delta_t*nsamples)/ntotatoms/6
+    onsager_std[1,1] = np.std(np.square([np.sum(disp_list_2[i]) for i in range(nsamples)]))/(delta_t*nsamples)/ntotatoms/6
+    return (onsager, onsager_std)
+
+def calc_directional_onsagers(disp_list_1, disp_list_2, delta_t, nsamples):
+    """ Calculate onsager coefficients for a binary alloy to yield
+        vectors of diffusivity in each direction for orientation analysis
+        Parameters
+        -----------
+        disp_list_1: list of ndarrays w/ shape (natoms,dims) if directional else (natoms,)
+        disp_list_2: list of ndarrays w/ shape (natoms,dims) if directional else (natoms,)
+        delta_t: float, time in picoseconds per traj frame
+        nsamples: int, number of samples for expectation values
+        -----------
+    """
+    # scaling factor
+    ntotatoms = len(disp_list_1[0][:,0]) + len(disp_list_2[0][:,0])
+    # x-direction average
+    onsager_xx = np.zeros((2,2))
+    onsager_xx[0,0] = np.average(np.square([np.sum(disp_list_1[i][:,0]) for i in range(nsamples)]))/(delta_t*nsamples)/ntotatoms/6

```

Comment:
maybe something like: np.sum(disp_list_1[:][:,0]) (not sure of syntax)

Maybe something like np.sum(disp_list_1[:][:,0], axis=?) ?